palette/src/rgb/rgb.rs

```diff
@@ -946,9 +948,13 @@ impl From<ParseIntError> for FromHexError {
 
 impl From<&'static str> for FromHexError {
     fn from(err: &'static str) -> FromHexError {
-        FromHexError::HexFormatError(err)
+        match err {
+            err if err.contains("rgba") => FromHexError::RgbaHexFormatError(err),
```

Comment:
It would be nice if we can avoid this and construct the error variant directly. I'm not so fond of relying on message content and would have suggested removing this `From` implementation if it wasn't a breaking change.

Reply:
Thanks for the feedback! I agree, I'll adjust it and push again.